src/main/java/seedu/address/model/loan/Nric.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.model.loan;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Class that stores the NRIC of a person.
+ */
+public class Nric {
+
+    public static final String MESSAGE_NRIC_CONSTRAINTS =
+            "Nric should be singapore issued. It may be blank. ";
+
+    public static final String NRIC_VALIDATION_REGEX = "^[STGF][0-9]{7}[A-Z]$";
+
+    public final String nric;
+
+    /**
+     * Constructs a {@code Nric}.
+     *
+     * @param ic A valid nric.
+     */
+    public Nric(String ic) {
+        ic = ic.trim().toUpperCase();
+        requireNonNull(ic);
+        checkArgument(isValidNric(ic), MESSAGE_NRIC_CONSTRAINTS);
+        nric = ic;
+    }
+
+    /**
+     * Returns true if a given string is a valid nric.
+     */
+    public static boolean isValidNric(String test) {
+        String ic = test.trim().toUpperCase();
```

Comment:
Same as above, do not do trimming here. The regex should reject it.

Reply:
We should still allow " s1234567X " even before trimming as it is valid. This allows the user input to be more flexible.

Reply:
Trimming is done at the command parser level (`ParserUtil.java`). I can agree with the mixed case but the other data fields do not accept strings with leading and trailing spaces as valid input.